pax-exam-container-carbon/org.wso2.carbon.container/src/main/java/org/wso2/carbon/container/CarbonTestContainer.java

```diff
@@ -0,0 +1,394 @@
+            }
+        });
+    }
+
+    /**
+     * Helper method to copy artifacts to the dropins.
+     *
+     * @param url             url of the artifact
+     * @param targetDirectory target directory
+     */
+    private void copyReferencedArtifactsToDeployDirectory(String url, Path targetDirectory) throws IOException {
+        File target = createUnique(url, targetDirectory.toFile());
+        FileUtils.copyURLToFile(new URL(url), target);
+    }
+
+    /**
+     * Create unique id to the file.
+     *
+     * @param url    url of the artifact
+     * @param deploy deploy directory
+     * @return file
+     */
+    private File createUnique(String url, File deploy) {
+        String prefix = UUID.randomUUID().toString();
+        String fileName = new File(url).getName();
+        return new File(deploy, prefix + "_" + fileName + ".jar");
+    }
+
+    /**
+     * Copy files specified in the carbon file copy option to the destination path.
+     *
+     * @param carbonHome carbon home
+     */
+    private void copyFiles(Path carbonHome) {
+        Arrays.asList(system.getOptions(CopyFileOption.class)).forEach(option -> {
+            try {
+                Files.copy(option.getSourcePath(), carbonHome.resolve(option.getDestinationPath()),
+                        StandardCopyOption.REPLACE_EXISTING);
+            } catch (IOException e) {
+                throw new TestContainerException("Error while copying configuration files", e);
+            }
+        });
+    }
+
+    /**
+     * Setup system properties.
+     *
+     * @param options options
+     * @throws IOException
+     */
+    private void setupSystemProperties(List<String> options, ExamSystem examSystem) throws IOException {
+        Arrays.asList(examSystem.getOptions(SystemPropertyOption.class)).forEach(systemPropertyOption -> {
+            String property = String.format("-D%s=%s", systemPropertyOption.getKey(), systemPropertyOption.getValue());
+            options.add(property);
+        });
+    }
+
+    /**
+     * Make all the files in the bin directory to be executable.
+     *
+     * @param carbonBin carbonBin
+     */
+    private void makeFilesInBinExec(File carbonBin) {
+        if (!carbonBin.exists()) {
+            return;
+        }
+        File[] files = carbonBin.listFiles();
+        if (files != null) {
+            Arrays.asList(files).forEach(file -> file.setExecutable(true));
+        }
+    }
+
+    /**
+     * Generate a random path if the unpack directory is not specified.
+     *
+     * @return the path to the carbon home directory
+     * @throws IOException
+     */
+    private Path retrieveFinalTargetDirectory() throws IOException {
+        Path unpackDirectory = carbonHomeDirectoryOption.getUnpackDirectory();
+        if (unpackDirectory == null) {
+            unpackDirectory = Paths.get("target", UUID.randomUUID().toString());
+        }
+        boolean isCreated = true;
```

Comment:
Try to initialize variables to their well-known defaults and build the logic
default for boolean is false.